Chained.h

```diff
@@ -2,17 +2,206 @@
+#include <string>
 
 template<typename Key, typename Value>
 class ChainedMap : public HashTable<Key, Value> {
-    public:
-        ChainedMap() = default;
+private:
+    double MAX_LOAD_FACTOR = 0.75;
 
-        Value get(Key k) = 0;
+    struct KeyValuePair{
+        KeyValuePair(Key k, Value v) {
+            key = k;
+            value = v;
+        }
+        Key key;
+        Value value;
+    };
 
-        void put(Key k, Value v) = 0;
+    std::vector<std::list<KeyValuePair>>* hashmap;
 
-        void remove(Key k) = 0;
+    std::hash<Key> hasher;
+
+    uint64_t hash(Key k){
+        return hasher(k)%hashmapSize;
+    }
+
+    uint64_t hashmapSize = 4;
+
+    uint64_t numPairs = 0;
+
+    void doubleSize(){
+        hashmapSize *= 2;
+        auto newMap = new std::vector<std::list<KeyValuePair>>(hashmapSize);
+        for(auto bucket : (*hashmap)){
+            for(auto kvp : bucket){
+                uint64_t index = hash(kvp.key);
+                (*newMap)[index].push_back(kvp);
+            }
+        }
+        auto temp = hashmap;
+        hashmap = newMap;
+        delete temp;
+    }
+
+    Value defaultGetReturn(){
+        std::cerr << "Invalid types for defaultGetReturn";
+        exit(1);
+    }
+
+    void printKeyValuePair(KeyValuePair kvp){
+        std::cout << " Invalid types for printing ";
+    }
+
+public:
+    ChainedMap() {
+        hashmap = new std::vector<std::list<KeyValuePair>>(hashmapSize);
+    }
+
+    ChainedMap(double load) {
+        hashmap = new std::vector<std::list<KeyValuePair>>(hashmapSize);
+        MAX_LOAD_FACTOR = load;
+    }
+
+    ~ChainedMap() {
+        delete hashmap;
+    }
+
+    Value get(Key k){
+        uint64_t index = hash(k);
+        for(auto kvp : (*hashmap)[index]){
+            if(k == kvp.key){
+                return kvp.value;
+            }
+        }
+        return defaultGetReturn();
+    }
+
+    void put(Key k, Value v) {
+        uint64_t index = hash(k);
+        for(auto kvp : (*hashmap)[index]){
+            if(kvp.key == k){
+                kvp.value = v;
+                return;
+            }
+        }
+        KeyValuePair kvp(k, v);
+        (*hashmap)[index].push_back(kvp);
+        ++numPairs;
+        if((static_cast<float>(numPairs))/(static_cast<float>(hashmapSize)) > MAX_LOAD_FACTOR){
+            doubleSize();
+        }
+    }
+
+    void remove(Key k) {
+        uint64_t index = hash(k);
+        for(auto i = (*hashmap)[index].begin(); i != (*hashmap)[index].end(); ++i) {
+            if(k == i->key){
+                (*hashmap)[index].erase(i);
+                --numPairs;
+                return;
+            }
+        }
+    }
+
+    void display(bool print){
+        std::vector<uint64_t> elements;
+        std::vector<uint64_t> collisions;
+        uint64_t numWith0 = 0;
+        uint64_t numWith1 = 0;
+        uint64_t numWithMore = 0;
+
+        //loop through and print the data structure while storing important information
```

Comment:
While collecting important information?

Why not have a `stats `method that computes and returns the stats?